frontend/src/component/FeedItem/Preview.jsx

```diff
@@ -236,28 +241,54 @@ export class Preview extends React.Component {
 
   getPreviewComponent () {
     const { props, state } = this
+    let component = null
     if (state.previewLoading) {
-      return <>{props.t('Preview loading...')}</>
+      component = <>{props.t('Preview loading...')}</>
     }
 
     if (state.previewUnavailable) {
-      return this.getUnavailablePreviewComponent()
-    }
-
-    if (this.isHtmlPreview()) {
+      component = this.getUnavailablePreviewComponent()
+    } else if (this.isHtmlPreview()) {
       if (state.previewHtmlCode === null) {
-        return this.getUnavailablePreviewComponent()
+        component = this.getUnavailablePreviewComponent()
+      } else {
+        component = this.getHTMLPreviewComponent()
       }
-
-      return this.getHTMLPreviewComponent()
+    } else {
+      component = this.getJPEGPreviewComponent()
     }
+    return (
+      <>
+        {component}
+        {state.previewOverflow && <div className='feedItem__preview__overflowOverlay' />}
+      </>
+    )
+  }
 
-    return this.getJPEGPreviewComponent()
+  getPreviewWithLink () {
+    const { props } = this
+    switch (props.linkType) {
+      case LINK_TYPE.NONE:
+        return this.getPreviewComponent()
+      case LINK_TYPE.OPEN_IN_APP:
+        return <Link to={props.link}>{this.getPreviewComponent()}</Link>
+      case LINK_TYPE.DOWNLOAD:
+        return (
+          <a
+            href={props.link}
+            download
+          >
+            {this.getPreviewComponent()}
+          </a>
+        )
+    }
+    return null
   }
 
   render () {
     const { props, state } = this
     const { content } = props
+    console.assert(props.linkType === LINK_TYPE.OPEN_IN_APP || content.contentType === CONTENT_TYPE.FILE)
```

Comment:
forgotten console?

Reply:
Not really, I put it here to warn future developers that this component cannot make a download link on something other than a file.

Reply:
Removed.